docs/developers/feature-versioning.md

```diff
@@ -0,0 +1,166 @@
+# Feature Versioning
+
+The stability levels of features (feature versioning) are independent of CRD API versions (API versioning).
+
+## Adding feature gates for API-driven features
+API-driven features are features that are enabled via a specific field in pipeline API. They comply to the [feature gates](../../api_compatibility_policy.md#feature-gates) and the [feature graduation process](../../api_compatibility_policy.md#feature-graduation-process) specified in the [API compatibility policy](../../api_compatibility_policy.md). For example, [remote tasks](https://github.com/tektoncd/pipeline/blob/454bfd340d102f16f4f2838cf4487198537e3cfa/docs/taskruns.md#remote-tasks) is an API-driven feature.
```

Comment:
NIT: I think "API-driven features" are features that are "accessed" or "consumed" via API, rather than "enabled".
Enabled is already associated with the flag, and I would find it consuming to use the same term.

Reply:
Done!